spec/lib/guardian/topic_guardian_spec.rb

```diff
@@ -195,6 +195,95 @@
     end
   end
 
+  describe "#can_create_topic?" do
+    before { SiteSetting.uncategorized_category_id = category.id }
+
+    context "when staff" do
+      it "always returns true and defaults to uncategorized if absent" do
+        expect(Guardian.new(moderator).can_create_topic?(category)).to eq(true)
+        expect(Guardian.new(moderator).can_create_topic?(nil)).to eq(true)
+      end
+    end
+
+    context "when trust level met" do
+      before { SiteSetting.min_trust_to_create_topic = 3 }
+
+      it "does not allow user to create the topic if they cannot create posts" do
+        guardian = Guardian.new(tl3_user)
+        guardian.stubs(:can_create_post?).with(topic).returns(false)
```

Comment:
I'm kind of curious about the use of stubs here because by stubbing, we're exposing the internal implementation details of the method to the test being written here. From my personal experience, this makes the test brittle when changes are made to the method in the future. Instead of stubbing, perhaps we can just use a topic that is in a category which the user has no access to?

Reply:
> we can just use a topic that is in a category which the user has no access to?

Doing this would make the subject under test (`can_create_topic?(topic)`) moot. 

The whole stubs vs not stub thing is tricky. If I write a test for `can_create_topic` that fulfils one case of `can_create_post`, but not the other, it is sort of an incomplete test, but I also don't want to have to rewrite the all the test cases of `can_create_post`... 🤔